tiatoolbox/tools/patchextraction.py

```diff
@@ -0,0 +1,330 @@
+        """Merge the patch-level results to get the overall image-level prediction.
+
+        Args:
+            patches: patch-level predictions
+
+        Returns:
+            image: merged prediction
+        """
+        raise NotImplementedError
+
+
+class FixedWindowPatchExtractor(PatchExtractor):
+    """Extract and merge patches using fixed sized windows for images and labels.
+
+    Args:
+        stride(Tuple of int): stride in (x, y) direction for patch extraction.
+
+    Attributes:
+        stride(Tuple of int): stride in (x, y) direction for patch extraction.
+
+    """
+
+    def __init__(
+        self,
+        input_img,
+        patch_size,
+        pad_size=0,
+        stride=1,
+    ):
+        super().__init__(
+            input_img=input_img,
+            patch_size=patch_size,
+            pad_size=pad_size,
+        )
+        self.stride = stride
+
+    # def extract_patches(self, input_img, labels=None):
+    #     raise NotImplementedError
+
+    def __next__(self):
+        raise NotImplementedError
+
+    def merge_patches(self, patches):
+        raise NotImplementedError
+
+
+class VariableWindowPatchExtractor(PatchExtractor):
+    """Extract and merge patches using variable sized windows for images and labels.
+
+    Args:
+        stride(Tuple of int): stride in (x, y) direction for patch extraction.
+        label_patch_size(Tuple of int): network output label (width, height).
+
+    Attributes:
+        stride(Tuple of int): stride in (x, y) direction for patch extraction.
+        label_patch_size(Tuple of int): network output label (width, height).
+
+    """
+
+    def __init__(
+        self,
+        input_img,
+        patch_size,
+        pad_size=0,
+        stride=1,
+        label_patch_size=None,
+    ):
+        super().__init__(
+            input_img=input_img,
+            patch_size=patch_size,
+            pad_size=pad_size,
+        )
+        self.stride_h = stride
+        self.label_patch_size = label_patch_size
+
+    # def extract_patches(self, input_img, labels=None):
+    #     raise NotImplementedError
+
+    def __next__(self):
+        raise NotImplementedError
+
+    def merge_patches(self, patches):
+        raise NotImplementedError
+
+
+class PointsPatchExtractor(PatchExtractor):
+    """Extracting patches with specified points as a centre.
+
+    Args:
+        labels(ndarray, pd.DataFrame, str, pathlib.Path): contains location and/or
+         type of patch. Input can be path to a csv or json file.
+        num_examples_per_patch(int): Number of examples per patch for ensemble
+         classification, default=9 (centre of patch and all the eight neighbours as
+         centre).
+
+    Attributes:
+        labels(pd.DataFrame): A table containing location and/or type of patch.
+        num_examples_per_patch(int): Number of examples per patch for ensemble
+         classification.
+
+    """
+
+    def __init__(
+        self,
+        input_img,
+        labels,
```

Comment:
I suggest using name `locations_list`, `annotations_list`, or 'points_list` instead of label. 

Also, it is good to accept the desired points list as a python `class 'list'`.

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/commit/0ad8ea23bbb8348cc1627fdd5aed5d30d838216c